src/vsphere_cpi/spec/integration/ip_conflict_detection_spec.rb

```diff
@@ -56,4 +57,115 @@
       end
     end
   end
+
+  context 'when testing with fully qualified network names' do
+    context 'when a VM already exists on network' do
+      context 'when the VM exists on a different network with same unqualified name' do
+
+        let(:network_spec_1) do
+          {
+            'static' => {
+              'ip' => fixed_ip_address,
+              'netmask' => '255.255.254.0',
+              'cloud_properties' => {'name' => vlan_1},
+              'default' => ['dns', 'gateway'],
+              'dns' => ['169.254.1.2'],
+              'gateway' => '169.254.1.3'
+            }
+          }
+        end
+
+        let(:network_spec_2) do
+          {
+            'static' => {
+              'ip' => fixed_ip_address,
+              'netmask' => '255.255.254.0',
+              'cloud_properties' => {'name' => vlan_2},
+              'default' => ['dns', 'gateway'],
+              'dns' => ['169.254.1.2'],
+              'gateway' => '169.254.1.3'
+            }
+          }
+        end
+
+        let(:vlan_1) { ENV.fetch('BOSH_VSPHERE_CPI_FOLDER_PORTGROUP_ONE') }
+        let(:vlan_2) { ENV.fetch('BOSH_VSPHERE_CPI_FOLDER_PORTGROUP_TWO') }
+
+        before do
+          verify_vlan(@cpi, vlan_1, 'BOSH_VSPHERE_CPI_FOLDER_PORTGROUP_ONE')
+          verify_vlan(@cpi, vlan_2, 'BOSH_VSPHERE_CPI_FOLDER_PORTGROUP_TWO')
+        end
+
+        it 'should create the VM' do
+          begin
+            test_vm_id = @cpi.create_vm(
+              'agent-007',
+              @stemcell_id,
+              vm_type,
+              network_spec_1,
+              [],
+              {'key' => 'value'}
```

Comment:
this should be empty hash, you are not using these values in this test, so no need to set it to anything